### PR TITLE
Full text / references / data drop down in results list

### PR DESCRIPTION
**Pull request by shinyichen**
* fixed orcid icon spacing in small screen menu
* Fix button size
* Sources dropdown in results list
Screen Shot 2021-10-29 at 3 30 52 PM

### CODE REVIEW

File: src/components/ResultList/Item/Item.tsx [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import { ItemType } from '@components/Dropdown/types';
 import { DatabaseIcon, DocumentTextIcon, ViewListIcon } from '@heroicons/react/outline';
+import { useViewport, Viewport } from '@hooks';
 import { getFomattedNumericPubdate } from '@utils';
 import { useMachine } from '@xstate/react';
 import clsx from 'clsx';
 import dynamic from 'next/dynamic';
 import Link from 'next/link';
-import { ReactElement } from 'react';
+import React, { ReactElement } from 'react';
```
Comment (thostetler) [inline src/components/ResultList/Item/Item.tsx:12]:
```suggestion
import { ReactElement } from 'react';
```

File: src/components/ResultList/Item/Item.tsx [hunk 2]
```diff
@@ -126,3 +247,27 @@ export const Item = (props: IItemProps): ReactElement => {
...
 };
+
+interface IItemDropdownProps {
+  label: ReactElement | string;
+  items: ItemType[];
+}
+
+export const ItemDropdown = ({ label, items }: IItemDropdownProps): ReactElement => {
+  const viewport = useViewport();
+  const listClasses = viewport > Viewport.MD ? 'h-auto w-auto' : 'h-auto w-auto absolute top-full -right-0';
```
Comment (thostetler) [inline src/components/ResultList/Item/Item.tsx:258]: If we're using this viewport calculation for just CSS, you should use tailwinds built-in utilities for that

https://tailwindcss.com/docs/responsive-design

File: src/components/ResultList/Item/Item.tsx [hunk 3]
```diff
@@ -66,6 +72,111 @@ export const Item = (props: IItemProps): ReactElement => {
...
+          className="default-icon default-link-color cursor-pointer"
+          aria-label="References and citations"
+          role="list"
+        />
+      ) : (
+        <ViewListIcon className="default-icon text-gray-300" aria-label="No references and citations" role="list" />
+      )}
+    </>
+  );
+
```
Comment (thostetler) [inline src/components/ResultList/Item/Item.tsx:179]: I think this unnecessarily complicates the `<Item/>` component, I think we can at least offload all this logic into a separate component or hook to clean this up a bit

**Review by thostetler — CHANGES_REQUESTED** [review]

File: src/components/ResultList/Item/Item.tsx (hunk 3 above)
Comment (shinyichen) [reply src/components/ResultList/Item/Item.tsx]: Moved the dropdowns into a separate component

File: src/components/ResultList/Item/Item.tsx (hunk 1 above)
Comment (shinyichen) [reply src/components/ResultList/Item/Item.tsx]: Removed

File: src/components/ResultList/Item/Item.tsx (hunk 2 above)
Comment (shinyichen) [reply src/components/ResultList/Item/Item.tsx]: Fixed.

**Comment by shinyichen** [comment]
@thostetler I fixed the issues mentioned. Please take another look.

**Pull request merged**